megamek/src/megamek/client/ui/swing/unitSelector/TWAdvancedSearchPanel.java

```diff
@@ -0,0 +1,1242 @@
+        techClassModel.addElement("All");
+        techClassModel.addElement("Inner Sphere");
+        techClassModel.addElement("Clan");
+        techClassModel.addElement("IS/Clan");
+        techClassModel.addElement("(Unknown Technology Base)");
+        techClassModel.setSelectedItem("All");
+        cboTechClass.setModel(techClassModel);
+        cboTechClass.addActionListener(this);
+
+
+        //Setup Weapons Table
+        scrTableWeapons.setMinimumSize(new Dimension(850, 150));
+        scrTableWeapons.setPreferredSize(new Dimension(850, 150));
+        weaponsModel = new WeaponsTableModel();
+        tblWeapons = new MegamekTable(weaponsModel, WeaponsTableModel.COL_NAME);
+        TableColumn wpsCol = tblWeapons.getColumnModel().getColumn(
+                WeaponsTableModel.COL_QTY);
+        wpsCol.setCellEditor(new DefaultCellEditor(cboQty));
+        tblWeapons.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        weaponsSorter = new TableRowSorter<>(weaponsModel);
+        tblWeapons.setRowSorter(weaponsSorter);
+        tblWeapons.addKeyListener(this);
+        TableColumn column = null;
+        for (int i = 0; i < WeaponsTableModel.N_COL; i++) {
+            column = tblWeapons.getColumnModel().getColumn(i);
+            if ((i == WeaponsTableModel.COL_QTY)) {
+                column.setPreferredWidth(40);
+            } else if ( i == WeaponsTableModel.COL_IS_CLAN) {
+                column.setPreferredWidth(75);
+            } else if ( i == WeaponsTableModel.COL_NAME) {
+                column.setPreferredWidth(310);
+            } else if ( i == WeaponsTableModel.COL_LEVEL) {
+                column.setPreferredWidth(100);
+            } else {
+                column.setPreferredWidth(50);
+            }
+        }
+        tblWeapons.setFont(new Font("Monospaced", Font.PLAIN, 12));
+        tblWeapons.getSelectionModel().addListSelectionListener(this);
+        scrTableWeapons.setViewportView(tblWeapons);
+
+        //Setup Equipment Table
+        scrTableEquipment.setMinimumSize(new java.awt.Dimension(850, 150));
+        scrTableEquipment.setPreferredSize(new java.awt.Dimension(850, 150));
+        equipmentModel = new EquipmentTableModel();
+        tblEquipment = new MegamekTable(equipmentModel,
+                EquipmentTableModel.COL_NAME);
+        TableColumn eqCol = tblEquipment.getColumnModel().getColumn(
+                EquipmentTableModel.COL_QTY);
+        eqCol.setCellEditor(new DefaultCellEditor(cboQty));
+        tblEquipment.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        equipmentSorter = new TableRowSorter<>(equipmentModel);
+        tblEquipment.setRowSorter(equipmentSorter);
+        tblEquipment.addKeyListener(this);
+        column = null;
+        for (int i = 0; i < EquipmentTableModel.N_COL; i++) {
+            column = tblEquipment.getColumnModel().getColumn(i);
+            if (i == EquipmentTableModel.COL_NAME) {
```

Comment:
This would be better handled as a switch.

Reply:
Same as above